src/Explorer/Controls/Settings/SettingsSubComponents/MongoIndexingPolicy/MongoIndexingPolicyComponent.tsx

```diff
@@ -0,0 +1,406 @@
+              notification={notification}
+              setRef={(textField: ITextField) => (this.currentTextFields[index] = textField)}
+              onIndexAddOrChange={(description: string, type: MongoIndexTypes) =>
+                this.props.onIndexAddOrChange(index, description, type)
+              }
+              onDiscard={() => {
+                this.currentTextFields.splice(index, 1);
+                this.props.onRevertIndexAdd(index);
+              }}
+            />
+          );
+        })}
+        <AddMongoIndexComponent
+          disabled={this.isIndexingTransforming()}
+          key={indexesToAddLength}
+          description={undefined}
+          type={undefined}
+          notification={undefined}
+          setRef={(textField: ITextField) => (this.currentTextFields[indexesToAddLength] = textField)}
+          onIndexAddOrChange={(description: string, type: MongoIndexTypes) =>
+            this.props.onIndexAddOrChange(indexesToAddLength, description, type)
+          }
+          onDiscard={() => {
+            this.props.onRevertIndexAdd(indexesToAddLength);
+          }}
+        />
+      </Stack>
+    );
+  };
+
+  private renderInitialIndexes = (): JSX.Element => {
+    const initialIndexes: MongoIndexDisplayProps[] = this.props.mongoIndexes
+      .map((mongoIndex: MongoIndex, arrayPosition: number) =>
+        this.getMongoIndexDisplayProps(mongoIndex, arrayPosition, true)
+      )
+      .filter((value: MongoIndexDisplayProps, index: number) => !!value && !this.props.indexesToDrop.includes(index));
+
+    return (
+      <Stack {...createAndAddMongoIndexStackProps} styles={mediumWidthStackStyles}>
+        <Stack
+          className="collapsibleLabel"
+          horizontal
+          tokens={accordionStackTokens}
+          onClick={this.toggleInitialIndexesExpanded}
+        >
+          <Icon
+            iconName={this.state.initialIndexesExpanded ? "ChevronDown" : "ChevronRight"}
+            styles={accordionIconStyles}
+          />
+          <Label>Current index(es)</Label>
+        </Stack>
+        {this.state.initialIndexesExpanded && (
+          <>
+            <DetailsList
+              styles={customDetailsListStyles}
+              disableSelectionZone
+              items={initialIndexes}
+              columns={this.initialIndexesColumns}
+              selectionMode={SelectionMode.none}
+              onRenderRow={this.onRenderRow}
+              layoutMode={DetailsListLayoutMode.justified}
+            />
+            {this.renderIndexesToBeAdded()}
+          </>
+        )}
+      </Stack>
+    );
+  };
+
+  private renderIndexesToBeDropped = (): JSX.Element => {
+    const indexesToBeDropped: MongoIndexDisplayProps[] = this.props.indexesToDrop.map(
+      (dropIndex: number, arrayPosition: number) =>
+        this.getMongoIndexDisplayProps(this.props.mongoIndexes[dropIndex], arrayPosition, false)
+    );
+
+    return (
+      <Stack styles={mediumWidthStackStyles}>
```

Comment:
Check if we can't reuse `CollapsiblePanel` in our code or if Fluent-UI has one. If we need a new one (probably), we should make this reusable and accessible. Use the pattern mentioned above (render `{this.props.children}`) so you can let the parent specify whatever component it wants inside this panel.